engine/runtime/src/main/java/org/enso/interpreter/runtime/type/Types.java

```diff
@@ -112,24 +112,29 @@ public static void extractArguments(Object[] arguments) throws ArityException {
    * @return the string representation of object's type.
    */
   public static String getName(Object value) {
-    if (TypesGen.isLong(value) || TypesGen.isImplicitLong(value)) {
-      return "Number";
+    if (TypesGen.isLong(value)
+        || TypesGen.isDouble(value)
+        || TypesGen.isEnsoBigInteger(value)
+        || TypesGen.isImplicitLong(value)) {
+      return "Builtins.Main.Number";
     } else if (TypesGen.isBoolean(value)) {
-      return "Boolean";
+      return "Builtins.Main.Boolean";
```

Comment:
I would extract these constants out into `TypeConstants` such that it can be used in tests _and_ here while avoiding duplication. Furthermore, you should note that this system is likely to change once we have support for shadow definitions as you'll have actual source files in which to refer to these builtins.